### PR TITLE
Node 4 syntax

### PR DESCRIPTION
**Pull request by SimenB**
Commits should probably be looked at individually. Can land stuff separately if you want?

What do you think about adding prettier, btw? Can simplify the eslint config as lots of things would be removed, and we'll probably end up with higher consistency on the code.

### CODE REVIEW

**Comment by SimenB** [comment]
I added a commit with prettier. I much prefer using that to eslint for style stuff, WDYT? 🙂

File: lib/register.js [hunk 1]
```diff
@@ -1,26 +1,4 @@
 'use strict';
```
Comment (SimenB) [inline lib/register.js]: this file can probably be deleted

**Comment by siimon** [comment]
Woah, huge PR :) gonna take awhile for me to review this :)

**Comment by siimon** [comment]
Do you know any alternatives to nock? Seems to be pretty unmaintained and I'm not sure I wanna merge this PR without green tests

**Comment by SimenB** [comment]
Tests on 4 and 6 passes, it's status quo wrt master.

And no, no alternatives at this time, looking for a replacement at work as well

**Comment by SimenB** [comment]
I can split it up if you want? The commits are pretty atomic

**Comment by SimenB** [comment]
@siimon I opened up #119, #120, #121, #122, #123, #124 and #125 with the changes in this PR split up. There'll be a lot of conflicts when merging, so if you press approve on the ones you think are good to go, I'll rebase the other ones as we go 🙂

**Comment by SimenB** [comment]
PR for node8: #126

**Comment by SimenB** [comment]
This has been split up, no need to keep it open

**Pull request closed without merging**